helpers/Physics.h

```diff
@@ -0,0 +1,380 @@
+					angY = BINANG_TO_RAD(rootAngleY);
+					pPrevRot->y = angY;
+				}
+			} else {
+				if (param->constraint.rotLimit.x) {
+					tempAngleX = RADF_TO_BINANG(rot[i - 2].x);
+					workAngleX = RADF_TO_BINANG(-Math_Atan2F(sqrtf(SQ(workVec.x) + SQ(workVec.z)), workVec.y));
+					Math_SmoothStepToS(&tempAngleX, workAngleX, 1, DEGF_TO_BINANG(param->constraint.rotLimit.x), 1);
+					angX = BINANG_TO_RAD(tempAngleX);
+					pPrevRot->x = angX;
+				}
+				if (param->constraint.rotLimit.y) {
+					tempAngleY = RADF_TO_BINANG(rot[i - 2].y);
+					workAngleY = RADF_TO_BINANG(Math_Atan2F(workVec.z, workVec.x));
+					Math_SmoothStepToS(&tempAngleY, workAngleY, 1, DEGF_TO_BINANG(param->constraint.rotLimit.y), 1);
+					angY = BINANG_TO_RAD(tempAngleY);
+					pPrevRot->y = angY;
+				}
+			}
+		} else {
+			angY = Math_Atan2F(workVec.z, workVec.x);
+			angX = -Math_Atan2F(sqrtf(SQ(workVec.x) + SQ(workVec.z)), workVec.y);
+			pPrevRot->y = angY;
+			pPrevRot->x = angX;
+		}
+		
+		Matrix_RotateY(angY, MTXMODE_NEW);
+		Matrix_RotateX(angX, MTXMODE_APPLY);
+		Matrix_MultZ(ABS(param->lengthList[i]) * param->gfx.scale.z, &posAdd);
+		
+		// Pushes limbs away from selected Vec3f points
+		for (s32 i = 0; i < param->spheres.numVec; i++) {
+			Vec3f tempPosAdd = {
+				pPrevPos->x + posAdd.x,
+				pPrevPos->y + posAdd.y,
+				pPrevPos->z + posAdd.z
+			};
+			f32 distTo = Math_Vec3f_DistXYZ(&tempPosAdd, &param->spheres.vecList[i]);
+			
+			if (distTo < param->spheres.dist) {
+				s16 yaw = Math_Vec3f_Yaw(&param->spheres.vecList[i], &tempPosAdd);
+				s16 pitch = Math_Vec3f_Pitch(&param->spheres.vecList[i], &tempPosAdd);
+				posAdd.x += Math_SinS(headRotY + 0x8000) * (param->spheres.dist - distTo) * 0.5f;
+				posAdd.z += Math_CosS(headRotY + 0x8000) * (param->spheres.dist - distTo) * 0.5f;
+				// posAdd.y += Math_SinS((-RADF_TO_BINANG(angX))) * (param->spheres.dist - distTo) * 0.5;
+				velAdj.x += Math_SinS(headRotY + 0x8000) * (param->spheres.dist - distTo) * 0.5f;
+				velAdj.z += Math_CosS(headRotY + 0x8000) * (param->spheres.dist - distTo) * 0.5f;
+				
+				pPrevRot->y = angY =  Math_Atan2F(posAdd.z, posAdd.x);
+				pPrevRot->x = angX = -Math_Atan2F(sqrtf(SQ(posAdd.x) + SQ(posAdd.z)), posAdd.y);
+			}
+		}
+		
+		workVec.x = pPos->x;
+		workVec.y = pPos->y;
+		workVec.z = pPos->z;
+		
+		pPos->x = pPrevPos->x + posAdd.x;
+		pPos->y = pPrevPos->y + posAdd.y;
+		pPos->z = pPrevPos->z + posAdd.z;
+		
+		pVel->x = (pPos->x - workVec.x + velAdj.x) * param->velMult;
+		pVel->y = (pPos->y - workVec.y + velAdj.y) * param->velMult;
+		pVel->z = (pPos->z - workVec.z + velAdj.z) * param->velMult;
+		
+		vel[i] = (Vec3f) {
+			CLAMP(pVel->x, -param->maxVel, param->maxVel),
+			CLAMP(pVel->y, -param->maxVel, param->maxVel),
+			CLAMP(pVel->z, -param->maxVel, param->maxVel),
+		};
+	}
+	
+	if (param->gfx.noDraw == true) {
+		Matrix_Pull();
+		
+		return;
+	}
+	
+	Gfx* disp = *dispType;
+	Mtx* matrix = Graph_Alloc(gfxCtx, param->J * sizeof(Mtx));
```

Comment:
```suggestion
	Mtx* matrix = Graph_Alloc(gfxCtx, (param->J - 1) * sizeof(Mtx));
```
i don't think the "last" Matrix "slot" is used?